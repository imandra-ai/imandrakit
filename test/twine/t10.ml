type foo =
  | A
  | B of { x: int }
[@@deriving twine, eq, show]

let x = B { x = 42 }
let x_str = Imandrakit_twine.Encode.encode_to_string foo_to_twine x
let x' = Imandrakit_twine.Decode.decode_string foo_of_twine x_str

let () =
  Format.printf "serialized %a is:@.%s@." pp_foo x
  @@ Imandrakit_twine.Dump.dump_string x_str

let () = assert (equal_foo x x')
