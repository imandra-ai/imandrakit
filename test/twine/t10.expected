serialized T10.B {x = 42} is:
[0x0]: C_1(42)

